packages/lab/src/list-next/ListNext.tsx

```diff
@@ -0,0 +1,169 @@
+import { makePrefixer, useForkRef, useId } from "@salt-ds/core";
+import {
+  Children,
+  cloneElement,
+  FocusEventHandler,
+  forwardRef,
+  HTMLAttributes,
+  isValidElement,
+  KeyboardEventHandler,
+  KeyboardEvent,
+  MouseEvent,
+  MouseEventHandler,
+  ReactElement,
+  FocusEvent,
+} from "react";
+import { clsx } from "clsx";
+import { ListItemNext as DefaultListItem } from "./ListItemNext";
+import { useList } from "./useList";
+import { useWindow } from "@salt-ds/window";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import listNextCss from "./ListNext.css";
+
+const withBaseName = makePrefixer("saltListNext");
+
+export interface ListNextProps extends HTMLAttributes<HTMLUListElement> {
+  /**
+   * If true, all items in list will be disabled.
+   */
+  disabled?: boolean;
+  /**
+   * The component used to render a ListItem instead of the default. This must itself render a ListItem,
+   * must implement props that extend ListItemProps and must forward ListItem props to the ListItem.
+   */
+  ListItem?: ReactElement;
+  /**
+   * Optional id prop
+   *
+   *The id for the list component
+   */
+  id?: string;
+  /**
+   * The number of items displayed in the visible area.
+   * Note that this determines the max height of the list.
+   */
+  displayedItemCount?: number;
+  // ListNextControlProps
+  onBlur?: FocusEventHandler;
+  onFocus?: FocusEventHandler;
+  onKeyDown?: KeyboardEventHandler;
+  onMouseDown?: MouseEventHandler;
+}
+
+export const ListNext = forwardRef<HTMLUListElement, ListNextProps>(
+  function ListNext(
+    {
+      children,
+      className,
+      disabled: listDisabled,
+      displayedItemCount: displayedItemCountProp = 4,
+      ListItem = DefaultListItem,
+      id: idProp,
+      onSelect,
+      onFocus,
+      onBlur,
+      onKeyDown,
+      onMouseDown,
+      style,
+      ...rest
+    },
+    ref
+  ) {
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-list-next",
+      css: listNextCss,
+      window: targetWindow,
+    });
+
+    const childrenCount = Children.count(children);
+
+    const listId = useId(idProp) || "listNext"; // TODO: check why useId needs to return undefined
```

Comment:
It needs to return undefined in the initial render due to the polyfill and ssr support.